tools/perfscale-load-generator/steady-state/steady-state.go

```diff
@@ -36,169 +34,170 @@ import (
+	b.createWatcherIfNotExist(objSample)
+
+	// Delete objects that match labels up to the count
+	objType := objUtil.GetObjectResource(objSample)
+	labels := objSample.GetLabels()
+	jobUUID := labels[config.WorkloadLabel]
+	log.Log.V(2).Infof("Deleting %d objects for job %s", count, jobUUID)
+	objUtil.DeleteNObjectsInNamespaces(b.virtClient, objType, config.GetListOpts(config.WorkloadLabel, jobUUID), count)
+
+	// Wait all objects be Deleted. In the case of VMI, deleted means the succeded phase.
+	for objType, objWatcher := range b.watchers {
+		log.Log.Infof("Wait for obj(s) %s to be deleted", objType)
+		objWatcher.WaitDeletion(b.Workload.Timeout.Duration)
 	}
-	wg.Wait()
+
+	log.Log.V(2).Infof("Deleted %d obj(s) for job %s", count, jobUUID)
 }
 
-func (b *SteadyStateJob) watchCreate(obj *unstructured.Unstructured) {
-	count := b.Workload.Count
-	objWatcher := watcher.NewObjListWatcher(
-		b.virtClient,
-		b.objType,
-		count,
-		*config.GetListOpts(config.WorkloadLabel, b.UUID))
-	objWatcher.Run()
-	log.Log.Infof("Wait for obj(s) %s to be available", b.objType)
-	objWatcher.WaitRunning(b.Workload.Timeout.Duration)
-
-	objWatcher.Stop()
+func (b *SteadyStateJob) DeleteNamespaces() {
+	log.Log.V(2).Infof("Clean up, deleting all created namespaces")
+	objUtil.CleanupNamespaces(b.virtClient, 30*time.Minute, config.GetListOpts(config.WorkloadLabel, b.UUID))
+	objUtil.WaitForDeleteNamespaces(b.virtClient, 30*time.Minute, *config.GetListOpts(config.WorkloadLabel, b.UUID))
 }
 
-func (b *SteadyStateJob) watchDelete(obj *unstructured.Unstructured) {
-	count := b.Workload.Count
-	// We expect b.churn fewer objects
-	count = count - b.churn
-
-	objWatcher := watcher.NewObjListWatcher(
-		b.virtClient,
-		b.objType,
-		count,
-		*config.GetListOpts(config.WorkloadLabel, b.UUID))
-	objWatcher.Run()
-	log.Log.Infof("Wait for obj(s) %s to be deleted", b.objType)
-	objWatcher.WaitDeletion(b.Workload.Timeout.Duration)
-
-	objWatcher.Stop()
+func (b *SteadyStateJob) createWatcherIfNotExist(objSpec *unstructured.Unstructured) {
+	objType := objUtil.GetObjectResource(objSpec)
+	if _, exist := b.watchers[objType]; !exist {
+		objWatcher := watcher.NewObjListWatcher(
+			b.virtClient,
+			objType,
+			b.Workload.Count,
+			*config.GetListOpts(config.WorkloadLabel, b.UUID))
+		objWatcher.Run()
+		b.watchers[objType] = objWatcher
+	}
 }
 
 func (b *SteadyStateJob) Wait() {
-	log.Log.V(1).Infof("SteadyState Load Generator Wait")
-	if b.minChurnSleep.Duration < time.Duration(1*time.Microsecond) {
-		time.Sleep(config.DefaultMinSleepChurn)
-		return
+	timePassed := time.Since(b.currTimestamp)
+	remainingTime := b.Workload.Timeout.Duration - timePassed
+	log.Log.V(3).Infof("Wait %f seconds before recreating %d objs. Remaining %f seconds to finish the test", b.minChurnSleep.Seconds(), b.Workload.Churn, remainingTime.Seconds())
+	time.Sleep(b.minChurnSleep.Duration)
+	// time.Sleep(remainingTime)
+	// b.currTimestamp = time.Now()
```

Comment:
Is this commented correctly? If yes, then maybe it should be removed and the `currTimestamp` should be renamed to `startedTime`?

Reply:
It's a leftover, I'll update it